vscode/oso/client/src/index.ts

```diff
@@ -113,106 +125,159 @@ async function openDocument(uri: Uri) {
   const uriMatch = (d: TextDocument) => d.uri.toString() === uri.toString();
   const doc = workspace.textDocuments.find(uriMatch);
   if (doc === undefined) await workspace.openTextDocument(uri);
+  return uri;
 }
 
-async function startClient(folder: WorkspaceFolder, context: ExtensionContext) {
-  const server = context.asAbsolutePath(join('out', 'server.js'));
-
-  // Watch `FileChangeType.Deleted` events for Polar files in the current
-  // workspace, including those not open in any editor in the workspace.
-  //
-  // NOTE(gj): Due to a current limitation in VS Code, when a parent directory
-  // is deleted, VS Code's file watcher does not emit events for matching files
-  // nested inside that parent. For more information, see this GitHub issue:
-  // https://github.com/microsoft/vscode/issues/60813. If that behavior is
-  // fixed in the future, we should be able to remove the
-  // `workspace.fileOperations.didDelete` handler on the server and go back to
-  // a single watcher for files matching '**/*.polar'.
-  const deleteWatcher = workspace.createFileSystemWatcher(
-    polarFilesInWorkspaceFolderPattern(folder),
-    true, // ignoreCreateEvents
-    true, // ignoreChangeEvents
-    false // ignoreDeleteEvents
-  );
-  // Watch `FileChangeType.Created` and `FileChangeType.Changed` events for
-  // files in the current workspace, including those not open in any editor in
-  // the workspace.
-  const createChangeWatcher = workspace.createFileSystemWatcher(
-    polarFilesInWorkspaceFolderPattern(folder),
-    false, // ignoreCreateEvents
-    false, // ignoreChangeEvents
-    true // ignoreDeleteEvents
-  );
+async function validateRoots(root: Uri, candidates: string[]) {
+  const projectRoots = [];
+  for (const path of [...new Set(candidates)]) {
+    // Split configured path on POSIX-style separator.
```

Comment:
Is that so that the configuration always uses `/`? I wonder if windows people would prefer using `\`

Reply:
Discussed offline; we're gonna ship it now with this POSIX bent since the configuration property validation & docs are very clear about it. If we hear from Windows users that this stinks, it probably wouldn't be too much more work to accept Windows and POSIX paths.